### ISSUE TITLE
[date-picker] 当date-picker组件type为datetime的format格式为YYYY年MM月DD日 HH时mm分ss秒选择面板输入框会出现秒会到第一个文本框后 日会到第二个文本框前

### ISSUE THREAD

**Issue by gitboyzcf** [element-plus/element-plus#14411]
Bug Type: **`Other`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.14`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkVtaXRzIERhdGUgb2JqZWN0PC9zcGFuPlxuICAgICAgPGRpdiBjbGFzcz1cImRlbW9uc3RyYXRpb25cIj5WYWx1ZToge3sgdmFsdWUxIH19PC9kaXY+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiUGljayBhIERhdGVcIlxuICAgICAgICBmb3JtYXQ9XCJZWVlZ5bm0TU3mnIhEROaXpSBISOaXtm1t5YiGc3Pnp5JcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG5cbjxzY3JpcHQ+XG5pbXBvcnQgeyByZWYgfSBmcm9tIFwidnVlXCI7XG5leHBvcnQgZGVmYXVsdCB7XG4gIHNldHVwKCkge1xuICAgIGNvbnN0IHZhbHVlMSA9IHJlZihcIlwiKTtcbiAgICByZXR1cm4ge1xuICAgICAgdmFsdWUxLFxuICAgIH07XG4gIH0sXG59O1xuPC9zY3JpcHQ+XG5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

点击时间日期输入框弹出面板》选择任意时间

### What is Expected?

面板上面两个输入框 比如
第一个输入框为 2023年09月27日  
第二个时间输入框为 00时00分00秒

### What is actually happening?

实际面板输入框内出现
第一个输入框为  2023年09月27日秒  
第二个时间输入框为  日 00时00分00秒

### Additional comments

_(empty)_

**Comment by StormerZSha** [comment]
这个好像是源码里用正则取format的一个问题
可以手动监听修改临时解决一下
```
watch(()=>value1.value,(cur)=>{
     let dom=document.querySelectorAll('.el-input--small .el-input__wrapper')
     let test=cur.split(' ')
     nextTick(()=>{
        for(let i=0;i<dom.length;i++){
          dom[i].children[0].value=test[i]
        }
    })
})
```

**Comment by FrontEndDog** [comment]
如果使用YYYY_MM_DD HH:mm:ss 也会有问题
![image](https://github.com/element-plus/element-plus/assets/46493087/a224fcb4-e71e-47e0-8bce-89defe38314d)
[例子](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkVtaXRzIERhdGUgb2JqZWN0PC9zcGFuPlxuICAgICAgPGRpdiBjbGFzcz1cImRlbW9uc3RyYXRpb25cIj5WYWx1ZToge3sgdmFsdWUxIH19PC9kaXY+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiUGljayBhIERhdGVcIlxuICAgICAgICBmb3JtYXQ9XCJZWVlZX01NX0REIEhIOm1tOnNzXCJcbiAgICAgIC8+XG4gICAgPC9kaXY+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuXG48c2NyaXB0PlxuaW1wb3J0IHsgcmVmIH0gZnJvbSBcInZ1ZVwiO1xuZXhwb3J0IGRlZmF1bHQge1xuICBzZXR1cCgpIHtcbiAgICBjb25zdCB2YWx1ZTEgPSByZWYoXCJcIik7XG4gICAgcmV0dXJuIHtcbiAgICAgIHZhbHVlMSxcbiAgICB9O1xuICB9LFxufTtcbjwvc2NyaXB0PlxuXG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl1cbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

**Comment by tolking** [comment]
Fixed by #14330
